### PR TITLE
trailing undescore for private and protected only

### PR DESCRIPTION
**Pull request by haianos**
Updating guidelines

### CODE REVIEW

File: cppguide.html [hunk 1]
```diff
@@ -4360,12 +4360,15 @@ <h4>Common Variable names</h4>
...
-<p>Data members of classes, both static and non-static, are
+<p>Private and protected data members of classes, both static and non-static, are
 named like ordinary nonmember variables, but with a
 trailing underscore.</p>
 
 <pre>class TableInfo {
   ...
+ protected:
+   <del>std::string ownerName; // NOT OK - trailing underscore missing.
+   std::string tableName_;
```
Comment (remod) [inline cppguide.html]:
```suggestion
 public:
  <del>std::string publicVariable_; // NOT OK - trailing underscore for public member variables.</del>
  <strong>std::string publicVariable; // OK - no trailing underscore for public member variables.</strong>
 protected:
  std::string protectedMember_;  // OK - underscore at end.
```

File: cppguide.html [hunk 2]
```diff
@@ -4360,12 +4360,15 @@ <h4>Common Variable names</h4>
 
 <h4>Class Data Members</h4>
 
-<p>Data members of classes, both static and non-static, are
+<p>Private and protected data members of classes, both static and non-static, are
```
Comment (remod) [inline cppguide.html]:
```suggestion
<p><del>Data</del><strong>Private and protected</strong> data members of classes, both static and non-static, are
```
Deviations from the official google style guide (which this guide is based on) are always marked by ~old~ **new**.

File: cppguide.html [hunk 3]
```diff
@@ -4374,20 +4377,23 @@ <h4>Class Data Members</h4>
 };
 </pre>
 
+<p><strong>Public</strong> data members of classes do not have the trailing underscore.</p>
```
Comment (remod) [inline cppguide.html]:
```suggestion
<p><strong>Public data members of classes do not have the trailing underscore.</strong></p>
```

File: cppguide.html [hunk 4]
```diff
@@ -4374,20 +4377,23 @@ <h4>Class Data Members</h4>
...
+
+<p>In short, the presence of the trailing underscore semantics indicate whether the member variable is public or not.</p>
```
Comment (remod) [inline cppguide.html]:
```suggestion
<p><strong>In short, the presence of the trailing underscore semantics indicate whether the member variable is public or not.</strong></p>
```

File: cppguide.html [hunk 5]
```diff
@@ -4374,20 +4377,23 @@ <h4>Class Data Members</h4>
...
+
 <h4>Struct Data Members</h4>
 
 <p>Data members of structs, both static and non-static,
```
Comment (remod) [inline cppguide.html]:
```suggestion
<p><strong>Data members of structs are named according to the same rule as data members of classes.</strong></p>
<p><del>Data members of structs, both static and non-static,
```

File: cppguide.html [hunk 6]
```diff
@@ -4374,20 +4377,23 @@ <h4>Class Data Members</h4>
...
-are named like ordinary nonmember variables, <strong>but with a
-trailing underscore.</strong> <del>They do not have
-the trailing underscores that data members in classes have.</del></p>
+are named like ordinary nonmember variables. They do not have
+the trailing underscores that private and protected data members in classes have, since they are all public.</p>
```
Comment (remod) [inline cppguide.html]:
```suggestion
are named like ordinary nonmember variables, do not have
the trailing underscores that data members in classes have.</del></p>
```

Members of structs are no all public, only if you don't put an access specifier. I would just go for `Data members of structs are named according to the same rule as data members of classes.`

WDYT?

File: cppguide.html [hunk 7]
```diff
@@ -4374,20 +4377,23 @@ <h4>Class Data Members</h4>
...
 
 <pre>struct UrlTableProperties {
   <del>std::string name;</del>
```
Comment (remod) [inline cppguide.html]:
```suggestion
  std::string name;
```

File: cppguide.html [hunk 8]
```diff
@@ -4374,20 +4377,23 @@ <h4>Class Data Members</h4>
...
-  <strong>std::string name_;</strong>
+  <strong>std::string name;</strong>
```
Comment (remod) [inline cppguide.html]:
```suggestion
```

File: cppguide.html [hunk 9]
```diff
@@ -4374,20 +4377,23 @@ <h4>Class Data Members</h4>
...
   <del>int num_entries;</del>
-  <strong>int numEntries_;</strong>
+  <strong>int numEntries;</strong>
   <del>static Pool&lt;UrlTableProperties&gt;* pool;</del>
```
Comment (remod) [inline cppguide.html]:
```suggestion
  static Pool&lt;UrlTableProperties&gt;* pool;
```

File: cppguide.html [hunk 10]
```diff
@@ -4374,20 +4377,23 @@ <h4>Class Data Members</h4>
...
-  <strong>static Pool&lt;UrlTableProperties&gt;* pool_;</strong>
+  <strong>static Pool&lt;UrlTableProperties&gt;* pool;</strong>
```
Comment (remod) [inline cppguide.html]:
```suggestion
```

**Comment by hogabrie** [comment]
LGTM, once @remod s inputs are incorporated.

**Comment by remod** [comment]
LGTM, should we merge or do you want to give more people the chance to vote?

**Pull request merged**